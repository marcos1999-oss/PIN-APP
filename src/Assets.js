export const logo = require("../assets/img/logo.png");
export const city = require("../assets/img/city.png");
export const city2 = require("../assets/img/city2.png");
export const city3 = require("../assets/img/city3.png");
export const cityMask = require("../assets/img/cityMask.png");
export const mapBG = require("../assets/img/map.jpg");
export const greenPin = require("../assets/img/green_pin.png");
export const currentLocation = require("../assets/img/current_location.png");
export const addProfile = require("../assets/img/add_profile.png");
export const bInfo = require("../assets/img/b_info.png");
export const bSettings = require("../assets/img/b_settings.png");
export const bSuppourt = require("../assets/img/b_suppourt.png");
export const bPrivacy = require("../assets/img/b_privacy.png");
export const bTerms = require("../assets/img/b_terms.png");
export const arrowBackWhite = require("../assets/img/arrow_back_white.png");
export const arrowFwdWhite = require("../assets/img/arrow_fwd_white.png");
export const alcohol = require("../assets/img/alcohol.png");
export const baby = require("../assets/img/baby.png");
export const driveThrough = require("../assets/img/drive_through.png");
export const heart = require("../assets/img/heart.png");
export const music = require("../assets/img/music.png");
export const parking = require("../assets/img/parking.png");
export const party = require("../assets/img/party.png");
export const pet = require("../assets/img/pet.png");
export const profile = require("../assets/img/profile.png");
export const reservation = require("../assets/img/reservation.png");
export const wifi = require("../assets/img/wifi.png");
export const valetParking = require("../assets/img/valet_parking.png");
export const manthCardRectangle = require("../assets/img/manthCard_rectangle.png");
export const userFavorite = require("../assets/img/user-favorite.png");
export const emiliano = require("../assets/img/example-profile.png");
export const heartDefault = require("../assets/img/heart-no-liked.png");
export const heartLiked = require("../assets/img/heart-is-liked.png");
export const mcDonald = require("../assets/img/mcDonald.png");
export const adBanner = require("../assets/img/jumbo-deal.png");
export const pinIcon = require("../assets/img/pinIcon.png");
export const locationIcon = require("../assets/img/locationIcon.png");
export const pin1 = require("../assets/img/pin1.png");
export const homePin = require("../assets/img/home-pin.png");
export const postLike = require("../assets/img/post-like.jpg");
export const postView = require("../assets/img/post-view.png");
export const shadow = require("../assets/img/shadow.png");
