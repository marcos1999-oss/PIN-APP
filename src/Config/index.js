import DebugConfig from './DebugConfig'

if (__DEV__) {
  console.disableYellowBox = !DebugConfig.yellowBox;
}

export default {
  BASE_URL: 'http://35.190.168.62/v1'
}