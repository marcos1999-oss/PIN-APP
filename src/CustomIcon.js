import {createIconSetFromIcoMoon} from 'react-native-vector-icons';
import icoConfig from './coustomIconMap.json';
export default createIconSetFromIcoMoon(icoConfig);