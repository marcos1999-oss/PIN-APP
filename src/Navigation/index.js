import React from "react";
import { BackHandler, Platform } from "react-native";
import { connect } from "react-redux";
import AppNavigation from "./AppNavigation";

class ReduxNavigation extends React.Component {
  componentWillMount() {
    if (Platform.OS === "android") {
      BackHandler.removeEventListener("hardwareBackPress", undefined);
    }
  }

  render() {
    return (
      <AppNavigation dispatch={this.props.dispatch} state={this.props.nav} />
    );
  }
}

const mapStateToProps = (state) => ({ nav: state.nav });

export default connect(mapStateToProps)(ReduxNavigation);
