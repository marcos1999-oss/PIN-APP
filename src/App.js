import React from "react";
import { BackHandler, Alert } from "react-native";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger'
import allReducers from './redux/reducers';
import creatSagaMiddleware from 'redux-saga';
import rootSaga from './redux/sagas/rootSaga';
import { Root } from "native-base";
import { DismissKeyboard } from "./Components/Common";
import firebase from 'react-native-firebase';
import AsyncStorage from '@react-native-community/async-storage';

/*Screen*/
import Splash from "./Screen/Splash";
import Navigation from "./Navigation";
if (__DEV__) {
  import('./reactotronConfig').then(() => console.log('Reactotron Configured'))
}

const sagaMiddleware = creatSagaMiddleware();
let store = createStore(allReducers, applyMiddleware(sagaMiddleware, logger));
sagaMiddleware.run(rootSaga);


class App extends React.Component {
  state = { isReady: false };

  constructor(props) {
    super(props);

    BackHandler.addEventListener("hardwareBackPress", async () => {
      await Alert.alert(
        "Exit Confirmation",
        "Sure you want to exit  ?",
        [
          {
            text: "OK",
            onPress: () => {
              BackHandler.exitApp();
            },
          },
          { text: "Cancel", style: "cancel" },
        ],
        { cancelable: true }
      );
    });
  }

  async checkPermission() {
    const enabled = await firebase.messaging().hasPermission();
    if (!enabled) {
      this.requestPermission();
    }
  }

  async requestPermission() {
    try {
      await firebase.messaging().requestPermission();
      // User has authorised
    } catch (error) {
      // User has rejected permissions
      console.log('permission rejected');
    }
  }

  async componentDidMount() {
    this.checkPermission();
    this.setState({ isReady: true });
  }

  //Remove listeners allocated in createNotificationListeners()
  componentWillUnmount() {
    this.notificationListener();
    this.notificationOpenedListener();
  }

  render() {
    if (!this.state.isReady) {
      return <Splash />;
    }

    return (
      <Provider store={store}>
        <DismissKeyboard>
          <Root>
            <Navigation />
          </Root>
        </DismissKeyboard>
      </Provider>
    );
  }
}

module.exports = { App, store };
