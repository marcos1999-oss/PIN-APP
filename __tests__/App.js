/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import mockComponent from 'react-native/jest/mockComponent';

import { App } from '../src/App';

jest.mock('../src/redux/services/config', () => {
  return {
    BASE_API_URL: 'http://127.0.0.1/v1'
  }
});

jest.mock('react-native-cached-images', () => {
  return {
    CustomCachedImage: mockComponent('Image'),
    CachedImage: mockComponent('Image'),
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {}
      }
    }
  }
});

jest.mock('@react-native-community/push-notification-ios', () => {
  return {
    addEventListener: jest.fn(),
    requestPermissions: jest.fn(),
    getApplicationIconBadgeNumber: jest.fn(),
    setApplicationIconBadgeNumber: jest.fn() 
  }
})

jest.mock('@hkpuits/react-native-region-monitor', () => {
  return {
    onRegionChange: jest.fn(),
    clearRegions: jest.fn(),
    addCircularRegion: jest.fn()
  }
})

jest.mock('react-native-geolocation-service', () => {
  return {
    getCurrentPosition: jest.fn(),
    watchPosition: jest.fn(),
  }
})

jest.mock('@react-native-community/geolocation', () => {
  return {
    getCurrentPosition: jest.fn(),
    watchPosition: jest.fn(),
  }
})

test('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
