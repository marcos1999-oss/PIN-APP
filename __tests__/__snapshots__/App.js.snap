// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={true}
      onLayout={[Function]}
      removeClippedSubviews={false}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "overflow": "hidden",
          },
        ]
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#5F92F3",
                              "flex": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {},
                              Object {
                                "backgroundColor": "transparent",
                              },
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "white",
                                "fontFamily": "SignPainter",
                                "fontSize": 63,
                                "transform": Array [
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                              }
                            }
                          >
                            scopin
                          </Text>
                        </View>
                      </View>
                      <Modal
                        animationType="fade"
                        hardwareAccelerated={false}
                        onRequestClose={[Function]}
                        transparent={true}
                        visible={false}
                      >
                        <View
                          accessible={true}
                          isTVSelectable={true}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "backgroundColor": "rgba(0,0,0,0.4)",
                              "flex": 1,
                              "justifyContent": "flex-end",
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            isTVSelectable={true}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "backgroundColor": "#fff",
                                "elevation": 4,
                                "maxHeight": 667,
                                "minHeight": 56,
                                "opacity": 1,
                                "padding": 15,
                              }
                            }
                          >
                            <RCTScrollView
                              data={Array []}
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              numColumns={1}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              style={
                                Array [
                                  Object {
                                    "marginHorizontal": -15,
                                  },
                                  Object {
                                    "marginTop": 0,
                                  },
                                ]
                              }
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                        </View>
                      </Modal>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </RNCSafeAreaView>
  <View
    style={
      Object {
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateY": -48,
          },
        ],
      }
    }
  />
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "maxHeight": 667,
            "minHeight": 56,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Array [
              Object {
                "marginHorizontal": -15,
              },
              Object {
                "marginTop": 0,
              },
            ]
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
